Shopify/shopify-cli
# Fix `shopify extension` commands timeout when organization has too many apps

### WHY are these changes introduced?

When users run `shopify extension` commands, and the organization has too many apps (~ 100), they're facing timeout issues:

<img width="80%" alt="error" src="https://user-images.githubusercontent.com/1079279/145858898-c5c53e5f-428b-44f7-9410-373b3e53b388.png">


### WHAT is this pull request doing?

Eager loading `extensionRegistrations` with `all_orgs_with_extensions.graphql` takes too long for organizations with a large number of apps.

I've initially considered paginating this query to avoid timeouts, but users would still need to wait too long.

So, this PR changes the `ShopifyCLI::PartnersAPI::Organizations.fetch_with_extensions` method to fetch `extensionRegistrations` individually and in parallel.

For a small number of apps, this change has almost no impact. But, for many apps (> 40), it gets expressive faster results:

<img width="80%" alt="compare" src="https://user-images.githubusercontent.com/1079279/145858909-1a2735ef-1a95-4117-9dc7-1176f0467469.png">

Also, this PR introduces a spinner in the `"Loading your extensions…"` message.

### How to test your changes?

Run `shopify extension connect` and notice the timeout won't happen anymore*:

<img width="80%" alt="compare" src="https://user-images.githubusercontent.com/1079279/145858902-493d5500-370f-4a85-8871-d6ecc665874b.gif">

> It takes some time to fetch 100 apps, so the Gif above demonstrates the command with 3 items. I've double-checked the mechanism locally with more than 100 apps, and I've not faced timeout issues.


### Post-release steps

None.

### Update checklist
